### PR TITLE
SnarkyJS parties uber test: Parties transaction through SnarkyJS/MinaSigner/Mina daemon.

### LINKED ISSUE #10438: Transaction construction end-to-end test in CI
A zk app that we deploy on a integration test network and send transactions with mina signer

In the test in the Mina repo

Maybe: `clone snarkyjs@alpha`, use local copy of signer; during the integration test

### LINKED ISSUE #10724: Enable more than one @method in snarkyjs

### LINKED ISSUE #10749: Fix invalid zkapp proofs in snarkyjs
Current hypothesis is that they're caused by using the wrong statement

### PR DESCRIPTION
**Pull request by bkase**
Explain your changes:
This PR adds a new integration test `snarkyjs.ml` along with a new buildkite job `"build-js-tests"` to support it. The test works as follows:
  * From the test executive, we call out to a nodejs script which can generate two transactions, one for deploying a zkapp and one for updating it with a proof
  * The nodejs script uses snarkyjs to generate the bulk of the transactions and mina-signer to add a feePayer and sign it. So we're testing the full "end-user" flow of interacting with zkapps
  * We send both of those transactions to the network and wait for them to be applied
  * We check that the new zkapp's balance and on-chain zkapp state were updated as expected

In addition to the above, there are some changes to the ocaml side of snarkyjs:
* Expose a `Sponge` to JS which can `absorb` and `squeeze` field elements. This was added a while ago in support of https://github.com/o1-labs/snarkyjs/pull/99, but we didn't have snarkyjs in develop back then, so it's merged here
* Enable the JS Pickles interface to compile SmartContracts with > 1 method. Closes https://github.com/MinaProtocol/mina/issues/10724
* Expose methods to compute and sign the transaction statement (in support of making the intg test work)
* In the mock Mina interface, check signatures before calling apply_parties_unchecked (in support of making the intg test work)

Explain how you tested your changes:
* By checking that the new integration test works in CI

Closes #10438
Closes #10749 
Closes https://github.com/o1-labs/snarkyjs/issues/110
Closes #10724

### CODE REVIEW

**Comment by mitschabaude** [comment]
!ci-build-me

**Review by mitschabaude — APPROVED** [review]

**Review by lk86 — APPROVED** [review]

File: src/app/test_executive/snarkyjs.ml [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            let zkapp_first_state =
+              Mina_base.Zkapp_state.V.to_list zkapp_account_update.app_state
+              |> List.hd_exn
+            in
+            let module Set_or_keep = Mina_base.Zkapp_basic.Set_or_keep in
+            let module Field = Snark_params.Tick0.Field in
+            let expected = Set_or_keep.Set (Field.of_int 3) in
+            if
+              Set_or_keep.equal Field.equal zkapp_first_state
+                (Set_or_keep.Set (Field.of_int 3))
```
Comment (deepthiskumar) [inline src/app/test_executive/snarkyjs.ml:175]: `expected`?

**Review by deepthiskumar — COMMENTED** [review]
There are some new changes in develop that aren't in this branch. Could you rebase against the latest develop before merging?

**Review by psteckler — CHANGES_REQUESTED** [review]
Please fill in the initial comment with the PR purpose and describing the testing that was done.

**Comment by mitschabaude** [comment]
I think we're good to go :D

**Review by deepthiskumar — APPROVED** [review]

**Review by psteckler — APPROVED** [review]

**Pull request merged**